test/unit/test_cluster_fuzz.py

```diff
@@ -0,0 +1,248 @@
+            if i >= 1 and i <= N:
+                assert os.path.exists(fuzz_file)
+                assert os.path.exists(flags_file)
+            else:
+                assert not os.path.exists(fuzz_file)
+                assert not os.path.exists(flags_file)
+
+    def test_fuzz_passes(self):
+        # We should see interesting passes being run in run.py. This is *NOT* a
+        # deterministic test, since the number of passes run is random (we just
+        # let run.py run normally, to simulate the real environment), so flakes
+        # are possible here. However, we do the check in a way that the
+        # statistical likelihood of a flake is insignificant. Specifically, we
+        # just check that we see a different number of passes run in two
+        # different invocations, which is enough to prove that we are running
+        # different passes each time. And the number of passes is on average
+        # over 100 here (10 testcases, and each runs 0-20 passes or so).
+        temp_dir = tempfile.TemporaryDirectory()
+        N = 10
+
+        # Try many times to see a different number, to make flakes even less
+        # likely. In the worst case if there were two possible numbers of
+        # passes run, with equal probability, then if we failed 100 iterations
+        # every second, we could go for billions of billions of years without a
+        # flake. (And, if there are only two numbers with *non*-equal
+        # probability then something is very wrong, and we'd like to see
+        # errors.)
+        seen_num_passes = set()
+        for i in range(100):
+            os.environ['BINARYEN_PASS_DEBUG'] = '1'
+            try:
+                proc = self.generate_testcases(N, temp_dir.name)
+            finally:
+                del os.environ['BINARYEN_PASS_DEBUG']
+
+            num_passes = proc.stderr.count('running pass')
+            print(f'num passes: {num_passes}')
+            seen_num_passes.add(num_passes)
+            if len(seen_num_passes) > 1:
+                return
+        raise Exception(f'We always only saw {seen_num_passes} passes run')
+
+    def test_file_contents(self):
+        # As test_fuzz_passes, this is nondeterministic, but statistically it is
+        # almost impossible to get a flake here.
+        temp_dir = tempfile.TemporaryDirectory()
+        N = 100
+        self.generate_testcases(N, temp_dir.name)
+
+        # To check for interesting wasm file contents, we'll note how many
+        # struct.news appear (a signal that we are emitting WasmGC, and also a
+        # non-trivial number of them), the sizes of the wasm files, and the
+        # exports.
+        seen_struct_news = []
+        seen_sizes = []
+        seen_exports = []
+
+        # The number of struct.news appears in the metrics report like this:
+        #
+        # StructNew      : 18
+        #
+        struct_news_regex = re.compile(r'StructNew(\s+):(\s+)(\d+)')
```

Comment:
Are the parens for readability? They shouldn't be necessary IIUC.

Reply:
Good point, we only need the last parens here, as that's all we actually read. Fixed.